### PR TITLE
MGMT-6550 Move SNO requiremets to versioned host requiremens configuration

### PR DESCRIPTION
**Pull request by jakub-dzon**
With this PR SNO requirements will start being defined as a separate `sno` property of `versioned-host-requirements` objects and corresponding default file will contain values previously specified for `HW_VALIDATOR_MIN_CPU_CORES_SNO` and `HW_VALIDATOR_MIN_RAM_GIB_SNO` env variables.

Signed-off-by: Jakub Dzon <jdzon@redhat.com>

### CODE REVIEW

**Comment by jakub-dzon** [comment]
https://github.com/openshift/assisted-test-infra/pull/837 needs to be merged first.

File: internal/bminventory/inventory.go [hunk 1]
```diff
@@ -4408,10 +4408,19 @@ func (b *bareMetalInventory) GetPreflightRequirements(ctx context.Context, param
 }
 
 func (b *bareMetalInventory) GetHostRequirements(_ context.Context, params installer.GetHostRequirementsParams) middleware.Responder {
```
Comment (tsorya) [inline internal/bminventory/inventory.go:4410]: i thought you removed this one already

**Comment by avishayt** [comment]
looks good to me

File: swagger.yaml [hunk 2]
```diff
@@ -3536,6 +3536,10 @@ definitions:
         x-go-name: WorkerRequirements
         $ref: '#/definitions/cluster-host-requirements-details'
         description: Worker node requirements
+      sno:
+        x-go-name: SNORequirements
+        $ref: '#/definitions/cluster-host-requirements-details'
+        description: SNO deployment node requirements
```
Comment (eranco74) [inline swagger.yaml:3542]:
```suggestion
        description: Single node OpenShift node requirements
```

File: data/default_hw_requirements.json [hunk 3]
```diff
@@ -11,5 +11,11 @@
     "ram_mib": 8192,
     "disk_size_gb": 120,
     "installation_disk_speed_threshold_ms": 10
+  },
+  "sno": {
```
Comment (eranco74) [inline data/default_hw_requirements.json:15]: I assume this is the file we maintain and all the rest are generated, correct?

File: config/onprem-iso-config.ign [hunk 4]
```diff
@@ -29,7 +29,7 @@
...
+          "source": "data:;base64,H4sIAAAAAAAC/7yVXW+jOBSG7/kZvgbzEQoUiQuaeBukTGCAdDRarSwXnMQawKxNstOt8t9XzlfT2czuaEbtBRI+B97H+DkSWVqU9zkqPs7wJC7ju7hAEevkQJqGCu2im8VF8SnNJxGpW9ZddhYFyo/VyR2epkUZ2Y4PLWhBW1WyNC+jG3fkqMXrh1Xoy2oef7iEFyh/SMYIqz3hRT6L1sPQy9A0iZRMDrQ2SM9gwyvSQN7TTq7ZcoCMh4HrjrQJymbpZ1zG+T0qI971grZaUaZ5fI+iJWuofJKDKh0p4xiPUV7iLC6nkVmTgZjdinVfDUmbpWSrjtawEoOWZmheTJPfSvyA8iJJ50X0DFzogfAZ1Ez2DXnCHWkpCFUV2h7QgaANJZLiLRWS8e5ai7Vkpd75c0OeIOPm+YOM4xNGTbcmr/rTOjxEGF8DD3uuSlpXXJ5zTmfVMiG4uDifirdmv3m8AGxds6Y97WraVYxKcx9kutBTFwwOa2N/f6QZDdtSeLiHTPIzXXA+LOVb4BXROMSfwe3qDL44WQ8GDnQsx7Id68YOHMuwgA7kpu+5GHBDt7QBIegFrzfVoN7Z6cCF/lV/Prz9jr7XnZ+x58PbN5Tnqwv6Z3k+9N9T3mv8j8vzYTBS8mxr5NzYnvu/8nRQ0yXZNAMIB7Ghe5fBVZcBtIxlBUffEXql/TNWTzFvprYX9IQzX3Bnz9/w30H2f27ox80HMHAP5m8sz7Gd0VXzj3QgYLfTssXdLBnjcTov42SOcpyj+6Qo8wQV0VGXlmQPHi4W2f7/o4ZDixflFJefMxR1vKPa9BN+iGfJJC5TFfBxkeToA5qXRfT7M3jZ2mnAdNASOVChpqvqN7jigkoQujoQpMUtewSh7Y0CV1ez9wVL9jfFK1V0LB0cf2tEDS0+9HtKazysBZVr3tS4lSC0rZ0O/uLiy78ozgUlsG+dX4XIjn9DCC4II8f3gl9D7P7Q/gkAAP//Uohy11wIAAA="
```
Comment (eranco74) [inline config/onprem-iso-config.ign:32]: @YuviGold any reason we have compression (gzip) here?
without it, the data will be more readable.

**Review by eranco74 — COMMENTED** [review]
/lgtm

**Comment by tsorya** [comment]
@jakub-dzon can you update https://github.com/openshift/assisted-service/blob/master/docs/dev/hardware-requirements.md too?

**Comment by jakub-dzon** [comment]
@tsorya update to the documentation is part of this PR: https://github.com/openshift/assisted-service/pull/1818/files#diff-9f60339c60872621c4d85b30a49006fe366055f1fc529ea1e6953ae7a38ef047

**Comment by jakub-dzon** [comment]
@eranco74 - could you, please, have a look again? The LGTM label from you has been removed after I pushed the requested changes.

File: config/onprem-iso-config.ign (hunk 4 above)
Comment (YuviGold) [reply config/onprem-iso-config.ign:32]: No idea.
@rwsu ?

**Pull request merged**